packages/SwingSet/docs/vat-environment.md

```diff
@@ -0,0 +1,113 @@
+
+This includes `Math`, but `Math.random` is disabled (calling it throws an error) as an obvious source of non-determinism. Likewise `Date` is included, but `new Date()` and `Date.now` throw an error. The remaining `Math` and `Date` features are purely computational and deterministic.
+
+Much of the `Intl` package, and some locale-specifc part of other objects (e.g. `Number.prototype.toLocaleString`) has results which depend upon which locale is configured, and that will vary from one process to another. The way we handle this is still in development. Either these functions will be disabled, or they will act as if they run on a host with a single fixed locale defined by the SES specification.
+
+As SES is on the JavaScript standards track, it includes other proposed features. If those features become standards, future JS environments will include them as global objects. So the current SES shim also makes those global objects available.
+
+## Additions
+
+`console` is available to vat code as a deliberate addition. It is not part of JavaScript, but it's so important for debugging that leaving it out would cause too much confusion. Note that the exact behavior of `console.log` is up to the host program: it is not guaranteed to be displayed to the operator. Applications that want to let vats write to `process.stdout` must somehow provide that authority directly. From a vat's point of view, `console` is frozen and powerless.
+
+`harden` is available as a global. It freezes the API surface (enumerable data properties) of an object. For more details, see [the `harden` package](https://github.com/Agoric/SES-shim/blob/SES-v0.8.0/packages/harden/README.md).
+
+`HandledPromise` is also a global. The `E` wrapper (`E(target).methodname(args)`) can be imported as `import { E } from '@agoric/eventual-send`. These are defined by the TC39 [Eventual-Send Proposal](https://github.com/tc39/proposal-eventual-send). In addition, "tildot" syntax (`target~.methodname(args)`) can be used by vat code.
+
+`Compartment`, [part of SES](https://github.com/Agoric/SES-shim/tree/SES-v0.8.0/packages/ses#compartment), is a global. Vat code runs inside a Compartment, but vat code can create sub-compartments to host other code (with different globals or transforms). Note that these child Compartments get `harden` and `Compartment`, but they won't include other endowments (like `console` or `HandledPromise`) unless you explicitly provide them. Child compartments will have the metering and tildot transformations applied (these are inescapable). Child compartments will *not* be frozen by default, so if you want to prohibit ambient communication channels through the global scope, you must call `harden(c.globalThis)` before loading any untrusted code into the new compartment.
```

Comment:
1) "`Compartment` (a [part of SES]()) is a global."
2) "inside a Compartment" needs `` around Compartment.
In general here, I think if it's Compartment, it's the specific type and should be ``ed. But for conceptual bits like "these child compartments" it should be lower-cased and not ``ed.
3) I'd start a new paragraph at "Note that..."
4) Is the "c" in c.globalThis a generic compartment? A keyword? Needs a quick mention of what it is.
5) "into the new `Compartment`".

Reply:
Yeah, my habit is to hold the `Compartment` instance in a variable named `c`. In this case, if I used `harden(compartment.globalThis)`, do you think that would be clear enough? I used to worry that the difference between `Compartment` (the constructor function) and `compartment` (an instance returned by some invocation ofthat function) was too subtle, but maybe not?

Reply:
how about:

Note that these child compartments get `harden` and `Compartment`, but they won't include other endowments (like `console` or `HandledPromise`) unless you explicitly provide them. Child compartments will have the metering and tildot transformations applied (these are inescapable). Child compartments will *not* be frozen by default, so if you want to prohibit ambient communication channels through the global scope, you must call `harden(compartment.globalThis)` after building `compartment` but before loading any untrusted code into it.

Reply:
re subtle difference between `Compartment` and `compartment`, try `compartment1`?

My coding style would be closer to `c`, but in documentation, the longer form seems worthwhile.